### PR TITLE
fix(Dataworker): Dereference originChainId by index, not key

### PR DESCRIPTION
**Pull request by pxrl**
Temporary fix to resolve an issue with future quoteTimestamps. We need a more robust solution than this in the long run.

### CODE REVIEW

**Comment by pxrl** [comment]
Hm; this is actually failing for me in local test, but I believe this fix is just unearthing a separate issue.

File: src/dataworker/PoolRebalanceUtils.ts [hunk 1]
```diff
@@ -84,17 +84,21 @@ export function updateRunningBalanceForEarlyDeposit(
   deposit: typechain.FundsDepositedEvent,
   updateAmount: BigNumber
 ): void {
+  // deposit.args is a pure array; there are no mapping to be dereferenced.
+  const originChainId = deposit.args[1].toNumber();
```
Comment (james-a-morris) [inline src/dataworker/PoolRebalanceUtils.ts:88]: We may want to wrap this as `Number(String(deposit.args[1]))` to ensure that we won't get a null reference. Alternatively - we can ensure that deposit.args[1] !== undefined

**Review by james-a-morris — COMMENTED** [review]
LGTM - pending linter

**Review by mrice32 — APPROVED** [review]

File: src/dataworker/PoolRebalanceUtils.ts (hunk 1 above)
Comment (mrice32) [reply src/dataworker/PoolRebalanceUtils.ts:88]: Will this work on a BN?

Comment (mrice32) [reply src/dataworker/PoolRebalanceUtils.ts:88]: I think .toString() will work on a BigNumber, not sure about String()

**Review by mrice32 — APPROVED** [review]
https://github.com/across-protocol/relayer-v2/pull/936#discussion_r1322129057

**Pull request merged**